enterprise/backend/src/metabase_enterprise/serialization/v2/utils/yaml.clj

```diff
@@ -24,15 +23,20 @@
       (str/replace "__SLASH__"     "/")
       (str/replace "__BACKSLASH__" "\\")))
 
-(defn hierarchy->file
-  "Given an extracted entity, return a [[File]] corresponding to it."
-  ^File [ctx entity]
+(defn hierarchy->path-components
+  "Given an extracted entity, return a seq of relative path components to it"
+  [ctx entity]
   (let [;; Get the desired [[serdes.base/storage-path]].
         base-path   (serdes.base/storage-path entity ctx)
         dirnames    (drop-last base-path)
         ;; Attach the file extension to the last part.
         basename    (str (last base-path) ".yaml")]
-    (apply io/file (:root-dir ctx) (map escape-segment (concat dirnames [basename])))))
+    (map escape-segment (concat dirnames [basename]))))
+
+(defn path-str
+  "Returns a human-readable path string from a sequence of path elements (like in :serdes/meta)"
+  [elements]
+  (->> elements (map (comp (partial str/join "#") (juxt :model :id))) (str/join "/")))
```

Comment:
It might be more readable to define a function formatting a path component for an element.
```clojure
(defn- format-path-component [element]
  (str (:model element) "#" (:id element)))
```